libcloud/common/azure.py

```diff
@@ -187,14 +187,17 @@ def _get_azure_auth_signature(self,
             else:
                 headers_copy[header] = value
 
+        is_change = method not in ("GET", "HEAD")
+        is_old_api = self.API_VERSION <= "2014-02-14"
+
         # Get the values for the headers in the specific order
         for header in special_header_keys:
             header = header.lower()  # Just for safety
             if header in headers_copy:
                 special_header_values.append(headers_copy[header])
-            elif header == "content-length" and method not in ("GET", "HEAD"):
-                # Must be '0' unless method is GET or HEAD
-                # https://docs.microsoft.com/en-us/rest/api/storageservices/authentication-for-the-azure-storage-services
+            elif header == "content-length" and is_change and is_old_api:
+                # For old API versions, the Content-Length header must be '0'
+                # https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key#content-length-header-in-version-2014-02-14-and-earlier
                 special_header_values.append('0')
```

Comment:
It wouldn't hurt to add a test case for it (we would need to assert inside the mock http class, which means we would also need to make MockHttp class inherit from unittest.TestCase, if not already) :)

Reply:
Added a test in 2eafc6d.